### PR TITLE
Update Travis config

### PR DESCRIPTION
**Pull request by rhcarvalho**
- Run checks with Go 1.6.x
- Run `go test` and `go build` in all packages
- Promote automatically gofmt'ing code instead of applying manual fixes based on diff output

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,6 @@
-
 language: go
-go: 
- - tip
-
+go:
+ - 1.6.3
```
Comment (philbrookes) [inline .travis.yml]: Didn't we agree to test both `tip` and `1.6.3`?

**Comment by rhcarvalho** [comment]
Example of passing build:
https://travis-ci.org/feedhenry/fh-system-dump-tool/builds/151450542

Example of build showing how tip is slow (requires building Go itself):
https://travis-ci.org/feedhenry/fh-system-dump-tool/builds/151449259

Example of build showing how release-branch.go1.6 is slow (requires building Go itself):
https://travis-ci.org/feedhenry/fh-system-dump-tool/builds/151447772

Example of failing build (after I intentionally added a file violating gofmt, and a failing test):
https://travis-ci.org/feedhenry/fh-system-dump-tool/builds/151451508

File: .travis.yml (hunk 1 above)
Comment (rhcarvalho) [inline .travis.yml]: We did, but see:
- https://github.com/feedhenry/fh-system-dump-tool/pull/19#issuecomment-239110882
- https://travis-ci.org/feedhenry/fh-system-dump-tool/builds/151449259

Making Travis build Go tip to run our tests wastes our time.

We use 1.6.3 to build releases, that should be the version we test against IMHO.
I initially left both tip and 1.6.3 in there, but I think I prefer the fastest feedback of just 1.6.3.

Do you still see strong value in running all PRs against Go tip?

Comment (philbrookes) [inline .travis.yml]: I see, I did not realize go was built for executing against tip, that seems excessive for less value. So in this case I'm happy to just test against 1.6.3.

**Comment by philbrookes** [comment]
:+1:

**Pull request merged**